pkgs/development/python-modules/ipywidgets/default.nix

```diff
@@ -1,48 +1,40 @@
-{ lib
-, buildPythonPackage
+{ buildPythonPackage
 , fetchPypi
-, python
-, nose
-, pytest
-, mock
-, ipython
 , ipykernel
+, ipython
 , jupyterlab-widgets
+, lib
+, nbformat
+, pytestCheckHook
 , traitlets
-, notebook
 , widgetsnbextension
 }:
 
 buildPythonPackage rec {
   pname = "ipywidgets";
   version = "7.7.0";
+  format = "setuptools";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "sha256-q0pVloVaiLg3YZIcdocH1l5YRwaBObwXKd3+g0cDVCo=";
+    hash = "sha256-q0pVloVaiLg3YZIcdocH1l5YRwaBObwXKd3+g0cDVCo=";
   };
 
-  # Tests are not distributed
-  # doCheck = false;
-
-  buildInputs = [ nose pytest mock ];
   propagatedBuildInputs = [
     ipython
     ipykernel
     jupyterlab-widgets
     traitlets
-    notebook
+    nbformat
     widgetsnbextension
   ];
 
-  checkPhase = ''
-    ${python.interpreter} -m unittest discover
-  '';
+  checkInputs = [ pytestCheckHook ];
 
-  meta = {
+  meta = with lib; {
```

Comment:
Please don't make these kind of changes. These are typically very personal. Others prefer using `with`, others not at all and want it removed from the language.

Reply:
Ah gotcha, I presumed that it would be ok based on the current usage of `with` here:
https://github.com/NixOS/nixpkgs/blob/541669b9b348181c846a20869590495a37ac8445/pkgs/development/python-modules/ipywidgets/default.nix#L46

As requested I've gone ahead and reverted this change in https://github.com/NixOS/nixpkgs/pull/171624/commits/fbccf6ead2e02e188daaa5a52bc98da4dc7679ea.

Reply:
It's OK to write it like this, the change is however not needed. You might make this change, and then the next contributor changes it to another style/preference, resulting in unnecessary code churn.

Reply:
Fair enough... do you feel comfortable with the latest version (https://github.com/NixOS/nixpkgs/pull/171624/commits/fbccf6ead2e02e188daaa5a52bc98da4dc7679ea)?